benches/event.rs

```diff
@@ -53,25 +60,31 @@ fn benchmark_event_create(c: &mut Criterion) {
     group.bench_function("single-level", |b| {
         b.iter(|| {
             let mut log = LogEvent::default();
-            log.insert("key1", Bytes::from("value1"));
-            log.insert("key2", Bytes::from("value2"));
-            log.insert("key3", Bytes::from("value3"));
+            log.insert(event_path!("key1"), Bytes::from("value1"));
+            log.insert(event_path!("key2"), Bytes::from("value2"));
+            log.insert(event_path!("key3"), Bytes::from("value3"));
         })
     });
 
     group.bench_function("nested-keys", |b| {
         b.iter(|| {
             let mut log = LogEvent::default();
-            log.insert("key1.nested1.nested2", Bytes::from("value1"));
-            log.insert("key1.nested1.nested3", Bytes::from("value4"));
-            log.insert("key3", Bytes::from("value3"));
+            log.insert(
+                event_path!("key1", "nested1", "nested2"),
+                Bytes::from("value1"),
+            );
+            log.insert(
+                event_path!("key1", "nested1", "nested3"),
+                Bytes::from("value4"),
+            );
+            log.insert(event_path!("key3"), Bytes::from("value3"));
         })
     });
     group.bench_function("array", |b| {
         b.iter(|| {
             let mut log = LogEvent::default();
-            log.insert("key1.nested1[0]", Bytes::from("value1"));
-            log.insert("key1.nested1[1]", Bytes::from("value2"));
+            log.insert(event_path!("key1", "nested1", 0), Bytes::from("value1"));
+            log.insert(event_path!("key1", "nested1", "1"), Bytes::from("value2"));
```

Comment:
This needs to be an integer to be treated as an index
```suggestion
            log.insert(event_path!("key1", "nested1", 1), Bytes::from("value2"));
```